### DISCUSSION TITLE
Symmetric commands: prepipe

### DISCUSSION THREAD

**Discussion by Poshi** [johnkerl/miller#938] [Ideas]
It feels weird to me that we have a `--prepipe[x]` flag but not a `--postpipe[x]` one. I know that you always can redirect output to the command of your choice though a pipe, but the same could be said about the input.
I can see the need of a `--prepipe[x]` command when we have more than a single input file, and usually many input files end up being a single output stream; but there's a case you are not covering (and is explicitly not supported according the documentation): in-place modification of files not compressed/coded in a standard way.

Let's say I'm using a hand-made compressor suited to my data that can reduce the size more than the standard compressors. Let's call it `reducer`. The tool to recover that data is called `expander`.
How would you perform an in-place modification of more than one of those files?
```
mlr --csv -I --prepipe expander --mfrom data1.csv.reduced data2.csv.reduced -- cut -f field1
```

With a `--postpipe[x]` command, that would be possible.

**Comment by johnkerl** [comment]
@Poshi you're right -- I've always thought we don't need post-pipe since we have the Unix pipe for output. But you're quite right -- `-I` is the right place for this. :)
